src/__tests__/spectral.test.ts

```diff
@@ -50,6 +51,85 @@ describe('spectral', () => {
     expect(expectedCustomRuleSet).toEqual(givenCustomRuleSet);
   });
 
+  // Assures: https://stoplightio.atlassian.net/browse/SL-787
+  test('given a ruleset with two identical rules under two distinct formats should not collide', () => {
+    const rulesets = [
+      {
+        rules: {
+          oas2: {
+            ruleName1: {
+              type: RuleType.STYLE,
+              function: RuleFunction.TRUTHY,
+              path: '$',
+              enabled: true,
+              summary: '',
+              input: {
+                properties: 'something-different',
+              },
+            },
+          },
+          oas3: {
+            ruleName1: {
+              type: RuleType.STYLE,
+              function: RuleFunction.NOT_CONTAIN,
+              path: '$.license',
+              enabled: false,
+              summary: '',
+              input: {
+                properties: ['url'],
+                value: 'gruntjs',
+              },
+            },
+          },
+        },
+      },
+    ];
+
+    const s = new Spectral({ rulesets });
+
+    expect(s.getRules('oas2')).toMatchInlineSnapshot(`
+Array [
+  Object {
+    "apply": [Function],
+    "format": "oas2",
+    "name": "oas2-ruleName1",
```

Comment:
We should probably remove the format from the name since format is included in the object and it seems repetitious